ssg/oval_object_model/oval_document.py

```diff
@@ -0,0 +1,307 @@
+                )
+            else:
+                if not OVALDocument._is_external_variable(component):
+                    # If OVAL entity is identical, but not external_variable, the
+                    # implementation should be rewritten each entity to be present
+                    # just once
+                    raise ExceptionDuplicateOVALEntity(
+                        "ERROR: OVAL ID "
+                        + component.id_
+                        + " is used multiple times"
+                        + " and should represent the same elements.\n Rewrite the OVAL checks."
+                        + " Place the identical IDs into their own definition and extend "
+                        + "this definition by it.\n"
+                    )
+
+    def finalize_affected_platforms(self, env_yaml):
+        """
+        Depending on your use-case of OVAL you may not need the <affected>
+        element. Such use-cases including using OVAL as a check engine for XCCDF
+        benchmarks. Since the XCCDF Benchmarks use cpe:platform with CPE IDs,
+        the affected element in OVAL definitions is redundant and just bloats the
+        files. This function removes all *irrelevant* affected platform elements
+        from given OVAL tree. It then adds one platform of the product we are
+        building.
+        """
+        type_ = required_attribute(env_yaml, "type")
+        full_name = required_attribute(env_yaml, "full_name")
+        for definition in self.definitions.values():
+            definition.metadata.finalize_affected_platforms(type_, full_name)
+
+    def load_definition(self, oval_definition_xml_el):
+        definition = load_definition(oval_definition_xml_el)
+        self._add_oval_component(definition, self.definitions)
+
+    def load_test(self, oval_test_xml_el):
+        test = load_test(oval_test_xml_el)
+        self._add_oval_component(test, self.tests)
+
+    def load_object(self, oval_object_xml_el):
+        object_ = load_object(oval_object_xml_el)
+        self._add_oval_component(object_, self.objects)
+
+    def load_state(self, oval_state_xml_element):
+        state = load_state(oval_state_xml_element)
+        self._add_oval_component(state, self.states)
+
+    def load_variable(self, oval_variable_xml_element):
+        variable = load_variable(oval_variable_xml_element)
+        self._add_oval_component(variable, self.variables)
+
+    def get_xml_element(self):
+        root = self._get_oval_definition_el()
+        root.append(self._get_generator_el())
+        root.append(self._get_definitions_el())
+        root.append(self._get_tests_el())
+        root.append(self._get_objects_el())
+        if self.states:
+            root.append(self._get_states_el())
+        if self.variables:
+            root.append(self._get_variables_el())
+        return root
+
+    def save_as_xml(self, fd):
+        root = self.get_xml_element()
+        ElementTree.indent(root, space=" ", level=0)
```

Comment:
Unfortunately, `indent()` is a new method in Python 3.9 and therefore we can't use it because we want to be compatible with 2.7 and 3.6. https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.indent

I think that you can make it optional by adding an if that is testing the presence of this method in the module and call it only if it exists.